alibi/explainers/ale.py

```diff
@@ -383,23 +398,27 @@ def ale_num(
         The constant offset used to center the ALE curves.
 
     """
-    if check_feature_resolution:
-        uniques = np.unique(X[:, feature])
-        if len(uniques) <= low_resolution_threshold:
-            q = uniques
+    if feature_grid_points is None:
+        if check_feature_resolution:
+            uniques = np.unique(X[:, feature])
+            if len(uniques) <= low_resolution_threshold:
+                q = uniques
+            else:
+                q, _ = adaptive_grid(X[:, feature], min_bin_points)
         else:
             q, _ = adaptive_grid(X[:, feature], min_bin_points)
-    else:
-        q, _ = adaptive_grid(X[:, feature], min_bin_points)
 
-    # if the feature is constant, calculate the ALE on a small interval surrounding the feature value
-    if len(q) == 1:
-        if extrapolate_constant:
-            delta = max(q * extrapolate_constant_perc / 100, extrapolate_constant_min)
-            q = np.hstack((q - delta, q + delta))
-        else:
-            # ALE is 0 at a constant feature value
-            return q, np.array([[0.]]), np.array([0.])
+        # if the feature is constant, calculate the ALE on a small interval surrounding the feature value
+        if len(q) == 1:
+            if extrapolate_constant:
+                delta = max(q * extrapolate_constant_perc / 100, extrapolate_constant_min)
+                q = np.hstack((q - delta, q + delta))
+            else:
+                # ALE is 0 at a constant feature value
+                return q, np.array([[0.]]), np.array([0.])
```

Comment:
I believe we would like to move this `if` statement outside as it can cover the case when `feature_grid_points` grid contains a singe point.